lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/cache/EvictableCache.java

```diff
@@ -178,13 +181,116 @@ public CacheStats stats()
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public boolean replace(K key, V oldValue, V newValue)
+            {
+                throw new UnsupportedOperationException("The operation is not supported, as in inherently races with cache invalidation");
+            }
+
+            @Override
+            public V replace(K key, V value)
+            {
+                throw new UnsupportedOperationException("The operation is not supported, as in inherently races with cache invalidation");
+            }
+
+            @Override
+            public int size()
+            {
+                return delegate.size();
+            }
+
+            @Override
+            public boolean isEmpty()
+            {
+                return delegate.isEmpty();
+            }
+
+            @Override
+            public boolean containsKey(Object key)
+            {
+                return delegate.containsKey(key);
+            }
+
+            @Override
+            public boolean containsValue(Object value)
+            {
+                for (Future<V> future : delegate.values()) {
+                    if (future.isDone() && !future.isCancelled() && Objects.equals(getDone(future), value)) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+            @Override
+            public V get(Object key)
+            {
+                return getIfPresent(key);
+            }
+
+            @Override
+            public V put(K key, V value)
+            {
+                throw new UnsupportedOperationException("The operation is not supported, as in inherently races with cache invalidation. Use get(key, callable) instead.");
+            }
+
+            @Override
+            public V remove(Object key)
+            {
+                Future<V> future = delegate.remove(key);
+                if (future != null && future.isDone() && !future.isCancelled()) {
+                    return getDone(future);
+                }
+                return null;
+            }
+
+            @Override
+            public void putAll(Map<? extends K, ? extends V> m)
+            {
+                throw new UnsupportedOperationException("The operation is not supported, as in inherently races with cache invalidation. Use get(key, callable) instead.");
+            }
+
+            @Override
+            public void clear()
+            {
+                delegate.clear();
+            }
+
+            @Override
+            public Set<K> keySet()
+            {
+                return delegate.keySet();
+            }
+
+            @Override
+            public Collection<V> values()
+            {
+                // values() should be a view, but also, it has a size and, iterating values shoulnd't throw for incomplete futures.
```

Comment:
shouldn't

Reply:
Should.  From JDK's docs for `java.util.Map#values`

> Returns a Collection view of the values contained in this map. The collection is backed by the map, so changes to the map are reflected in the collection, and vice-versa.

Reply:
I'm refering to typo :)

Reply:
thanks!